resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -542,4 +728,114 @@ private void addTrace(Throwable cause) {
         trace.append("Caused by: ");
         trace.append(cause);
     }
+
+    /**
+     * Holds the closed DNS Servers for a domain.
+     */
+    private static final class DnsServerNames {
+
+        private final String qName;
+
+        // We not expect the linked-list to be very long so a double-linked-list is overkill.
+        private ServerName head;
+        private int count;
+
+        DnsServerNames(String qName) {
+            this.qName = qName.toLowerCase(Locale.US);
+        }
+
+        void add(DnsRecord r) {
+            if (r.type() != DnsRecordType.NS || !(r instanceof DnsRawRecord)) {
+                return;
+            }
+
+            // Only include servers that serve the correct domain.
+            if (qName.length() <  r.name().length()) {
+                return;
+            }
+
+            String rName = r.name().toLowerCase(Locale.US);
```

Comment:
Does `rName` stand for `recordName` ... or some other concept? If this is not a special concept in the RFC can we use more descriptive variable names?